nixos/modules/services/backup/duplicity.nix

```diff
@@ -99,18 +122,24 @@ in {
 
         environment.HOME = stateDirectory;
 
-        serviceConfig = {
-          ExecStart = ''
-            ${pkgs.duplicity}/bin/duplicity ${escapeShellArgs (
-              [
-                cfg.root
-                cfg.targetUrl
-                "--archive-dir" stateDirectory
-              ]
+        script =
+          let
+            target = escapeShellArg cfg.targetUrl;
+            extra = escapeShellArgs ([ "--archive-dir" stateDirectory ] ++ cfg.extraFlags);
+            dup = "${pkgs.duplicity}/bin/duplicity";
+          in
+          ''
+            set -x
+            ${dup} cleanup ${target} --force ${extra}
```

Comment:
Will all extra parameters that can be passed to 'duplicity incr' fit 'duplicity cleanup', remove-older-than, or remove-all-but-n-full?

Reply:
In this case, instead of maxAge and maxFull, what do you think of an option to set the list of actions, the default being ["incr"].

Reply:
verbs don't take the same positional arguments: cleanup only takes the target, and incr takes both a source and a target, so there seems to be no way to provide such a unified interface.

Reply:
OK. We could add the source only for verbs that need it, but on the other hand the only verbs that make sense in a timer are:
- full
- incr
- remove-older-than
- remove-all-but-n-full
- remove-all-inc-of-but-n-full

so if you add an option for remove-all-inc-of-but-n-full we're covered.
(full is incr --full-if-older-than 0, probably.)

What do you think?

Reply:
I'd rather add options `maxIncAge` for remove-all-inc-but-n-full and `mode = "full" | "incremental"` to choose between incremental and full backups (and maybe an option to set  `--full-if-older-than`).

This way you specify the end result you want instead of the list of actions to take. Also it's kind of self documenting by reading the option list.

Reply:
I just did that, please have a look.